pkg/statistics/handle/storage/read.go

```diff
@@ -529,9 +529,12 @@ func TableStatsFromStorage(sctx sessionctx.Context, snapshot uint64, tableInfo *
 	if err != nil {
 		return nil, err
 	}
-	// Check deleted table.
+	// Check table has no index/column stats.
 	if len(rows) == 0 {
-		return nil, nil
+		if table.ColAndIdxExistenceMap.ColNum()+table.ColAndIdxExistenceMap.IdxNum() > 0 {
+			table.CleanUpStats()
+		}
```

Comment:
It can not solve the edge cases in the old version cluster. May need another solution.